pvlib/iotools/pvgis.py

```diff
@@ -510,6 +533,9 @@ def get_pvgis_tmy(latitude, longitude, outputformat='json', usehorizon=True,
     if map_variables:
         data = data.rename(columns=VARIABLE_MAP)
 
+    if utc_offset != 0:
+        data = _coerce_and_roll_pvgis_tmy(data, utc_offset, coerce_year)
```

Comment:
I did recognize that forcing year could be useful by itself (eg when utc is zero)  but figured users could more easily replace the index with time series of their choice since no rolling is required. Eg:
```python
data.index = pd.date_range(start='year-01-01 01:00', freq='T', …)
```
I didn’t want to make `utc_offset` be either boolean or integer, because I’m afraid it’s a risky design that later will cause unexpected problems, eg zero is also false. I also didn’t want to add a 3rd new argument, so that’s why I left it like this.

Reply:
Would something like the following work?

```python
def get_pvgis_tmy(..., roll_offset=0, coerce_year=1990):
    """
    Parameters
    ----------
    ...
    roll_offset : int, default 0
        A UTC offset in hours.  If provided, the dataframe will be "rolled"
        so that data begins at midnight, blah blah
        It is likely desirable to use this parameter in combination with
        ``coerce_year``.
    coerce_year : int, default 1990
        blah blah
    """

    if roll_offset != 0:
        tz = pytz.timezone(f'Etc/GMT{-roll_offset:+d}')
        data = data.tz_convert(tz)
        data = pd.DataFrame(np.roll(data, roll_offset, axis=0),
                            columns=data.columns,
                            index=np.roll(data.index, roll_offset))

    if coerce_year:
        data.index = [x.replace(year=1990) for x in data.index]
```

Reply:
I don’t think this will work. When I tried this variant I got an error that February 29 is not a date. The `tz_convert()` changed the hour which made March 1st at midnight become 1am and made midnight now February 29th. I don’t want to have to sanitize the February year PVGIS chooses before converting. Look back at 31800e9 and d0bd0a0.

Reply:
Argh, good point.  Shame on me for not actually trying it.  

I still think it's strange for `coerce_year` to only work when `utc_offset` is nonzero.  Does your code still work if it runs `_coerce_and_roll_pvgis_tmy` even when `utc_offset` is zero?  Seems like it should...

Reply:
I can make it work - give me a second

Reply:
> Argh, good point. Shame on me for not actually trying it.
> 
> I still think it's strange for `coerce_year` to only work when `utc_offset` is nonzero. Does your code still work if it runs `_coerce_and_roll_pvgis_tmy` even when `utc_offset` is zero? Seems like it should...

`align_to_local_noon` might be less cryptic.

Reply:
@kandersolar ready for review -- see my comment (updated) and my last 2 commits.